charms/worker/k8s/src/charm.py

```diff
@@ -1340,6 +1358,43 @@ def _on_refresh_certs_action(self, event: ops.ActionEvent):
             except (InvalidResponseError, K8sdConnectionError) as e:
                 event.fail(f"Failed to refresh certificates: {e}")
 
+    @on_error(
+        ops.BlockedStatus("Cannot change bootstrap config after bootstrap. Check logs."),
+        BootstrapConfigChangeError,
+    )
+    @on_error(
+        ops.ErrorStatus("Failed to get communicate with k8sd."),
```

Comment:
I don't think its really meant for the charm to use `ops.ErrorStatus`.  Usually this is reserved for the jujud agent to raise.  @mateoflorido could correct me but i think the charm should only ever raise.  That's because an error condition tells the user they MUST `juju resolve <unit>` in order for it to execute anymore hooks.  Raising an error essentially is the charm giving up working until the user intervenes -- that could be years. 

stick to: maintenance, active, waiting, and blocked.  

In which case -- this COULD be blocked.  OR it could be waiting.  K8sd didn't answer?  My guess is it would on the next hook.  Go with `Waiting` here.